ffs/filcold/filcold.go

```diff
@@ -260,12 +257,8 @@ func (fc *FilCold) WaitForDeals(ctx context.Context, c cid.Cid, proposals []cid.
 			break
 		}
 	}
-
-	if len(activeProposals) == 0 {
-		if ctx.Err() != nil {
-			return nil, errors, fmt.Errorf("all accepted proposals were untracked due to cancellation")
-		}
-		return nil, errors, fmt.Errorf("all accepted proposals failed before becoming active")
```

Comment:
Same here. `WaitForDeals` just wait for deals until final status. 
If all didn't reach `Success` or whatever, isn't something related to `WaitForDeals`. `WaitForDeals` waits... then the interpretation of "failure" is up to the caller.
All this makes simpler to use these APIs... no "border"cases error cases or similar.